File: VB/AutoScrollTimer/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms

Namespace AutoScrollTimer

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private autoScrollHelper As AutoScrollHelper

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            gridControl1.DataSource = GetDataTable()
            autoScrollHelper = New AutoScrollHelper(gridView1)
        End Sub

        Private Function GetDataTable() As DataTable
            Const ColCount As Integer = 40
            Const RowCount As Integer = 100
            Dim table As DataTable = New DataTable()
            For i As Integer = 0 To ColCount - 1
                table.Columns.Add()
            Next

            For j As Integer = 0 To RowCount - 1
                Dim row As DataRow = table.NewRow()
                For i As Integer = 0 To ColCount - 1
                    row(i) = String.Format("row {0} / col {1}", j, i)
                Next

                table.Rows.Add(row)
            Next

            table.AcceptChanges()
            Return table
        End Function

        Private Sub simpleButton1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs)
            simpleButton1.DoDragDrop("test", DragDropEffects.Move)
        End Sub

        Private Sub gridControl1_DragOver(ByVal sender As Object, ByVal e As DragEventArgs)
            e.Effect = DragDropEffects.Move
            autoScrollHelper.ScrollIfNeeded()
        End Sub

        Private Sub gridControl1_DragLeave(ByVal sender As Object, ByVal e As EventArgs)
        End Sub
    End Class
End Namespace

File: CS/AutoScrollTimer/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;

namespace AutoScrollTimer {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        AutoScrollHelper autoScrollHelper;
        private void Form1_Load(object sender, EventArgs e) {
            gridControl1.DataSource = GetDataTable();
            autoScrollHelper = new AutoScrollHelper(gridView1);
        }

        private DataTable GetDataTable() {
            const int ColCount = 40;
            const int RowCount = 100;
            DataTable table = new DataTable();
            for(int i = 0; i < ColCount; i++)
                table.Columns.Add();
            for(int j = 0; j < RowCount; j++) {
                DataRow row = table.NewRow();
                for(int i = 0; i < ColCount; i++)
                    row[i] = string.Format("row {0} / col {1}", j, i);
                table.Rows.Add(row);
            }
            table.AcceptChanges();
            return table;
        }

        private void simpleButton1_MouseMove(object sender, MouseEventArgs e) {
            simpleButton1.DoDragDrop("test", DragDropEffects.Move);
        }

        private void gridControl1_DragOver(object sender, DragEventArgs e) {
            e.Effect = DragDropEffects.Move;
            autoScrollHelper.ScrollIfNeeded();
        }

        private void gridControl1_DragLeave(object sender, EventArgs e) {
            
        }
    }
}